fe/fe-core/src/test/java/com/starrocks/analysis/AlterUserStmtTest.java

```diff
@@ -143,4 +145,21 @@ public void testInvalidAuthPlugin() throws Exception {
         UtFrameUtils.parseStmtWithNewParser(sql, ConnectContext.get());
         Assert.fail("No exception throws.");
     }
+
+    @Test(expected = SemanticException.class)
+    public void testRootPassword() throws Exception {
+        String createUserSql = "CREATE USER 'admin' IDENTIFIED BY ''";
+        CreateUserStmt createUserStmt =
+                (CreateUserStmt) UtFrameUtils.parseStmtWithNewParser(createUserSql, starRocksAssert.getCtx());
+        AuthenticationMgr authenticationManager =
+                starRocksAssert.getCtx().getGlobalStateMgr().getAuthenticationMgr();
+        authenticationManager.createUser(createUserStmt);
+
+        starRocksAssert.getCtx().setCurrentUserIdentity(new UserIdentity("admin", "%"));
+        starRocksAssert.getCtx().setCurrentRoleIds(new UserIdentity("admin", "%"));
+
+        AlterUserStmt alterUserStmt = (AlterUserStmt) UtFrameUtils.parseStmtWithNewParser(
+                "alter user root identified by \"12345\"; ", starRocksAssert.getCtx());
```

Comment:
Have you tried if the statement `SET PASSWORD for ROOT ` can be prohibited now?

Reply:
Yes, it is because set password is not allowed, so a unified one is needed here